### ISSUE TITLE
Feature Request: Key binding

### ISSUE THREAD

**Issue by wortmannkd** [Timmeey86/FS25_UnloadBalesEarly#13]
Hi Timmeey86! Thank you for the mod that early in FS25 history!

Apparently the key binding for unloading unfinished bales is hard coded, am I right?

I'm playing FS25 primarily with a XBox controller atm and have two key bindings in the F1 menue for "Unload unfinished bale early". One is the standard key binding for unloading (finished) bale (RB + Y) and additionally there is LS + DPdown. As I accidentally unloaded unfinished bales by mistake I wanted to reset the default RB + Y to not take effect for the mod but only for the ingame standard (unloading finished bales) and only use the second LS + DPdown I figured there is no key binding option for this mod in the key binding screen but think that would be great (in my case especially for avoiding unloading unfinished bales by mistake via the standard key binding RB + Y).

Long story short: As long this isn't a bug on my side I would like to suggest to implement the possibilty to set the key bindings – if you like and if it's possible.

Thanks for reading!

**Comment by Timmeey86** [comment]
Hi, I'll try implementing this, it might take a while

**Comment by Timmeey86** [comment]
[FS25_UnloadBalesEarly.zip](https://github.com/user-attachments/files/18047827/FS25_UnloadBalesEarly.zip)


Here you go

**Comment by wortmannkd** [comment]
Thank you so much, Timmeey86 for taking a look into it. It works nicely so that you can assign individual keys to the function. Great!
I don't want to come off picky but the action to unload an unfinished bale is still as well set to the default keys. I assume it is bound to the default key binding of unloading bales from basegame. Would it be possible to set the default key binding of the mod to the same? So that when a user changes the key bindings in the key binding screens it won't be left on the default key binding (RB + Y)?

It's just a question. This is you mod of course. I can live with it so I have to learn to be careful when having less than 100 % in the baler. :D But thanks anyway!!

**Comment by Timmeey86** [comment]
I think the default binding just uses "Tool Function 4". I don't think it's a good idea for a mod to change the binding for Tool Function 4 since there are also other things bound to that by the base game
